interfaces/cython/cantera/speciesthermo.pyx

```diff
@@ -72,6 +79,9 @@ cdef class SpeciesThermo:
                                            T_high, P_ref, &data[0])
             return data
 
+    def _check_n_coeffs(self, n):
+        raise NotImplementedError('Needs to be overloaded')
```

Comment:
A docstring for this function would be useful.

Also, we could avoid the need to override this in every child class by having `return n == self.n_coeffs` be the default implementation, which would be correct for all classes where the number of coefficients is a constant.

Reply:
Ok on the docstring.

The other suggestion won’t work all the time as the number of zones is not known a priori for the NASA9 multi temp case.

Reply:
Ok to docstring.

The suggestion won't work as the function is called prior to the instantiation of the C++ object. This is clarified in the docstring.

Reply:
Ah, fair point. The documentation of the tricky behavior is appreciated.